import "comm_message.proto";

//排行榜类型
enum RankType
{
	//等级排行
	RANK_LEVEL_TOTAL = 11;
	RANK_LEVEL_JOB1 = 12;
	RANK_LEVEL_JOB2 = 13;
	RANK_LEVEL_JOB3 = 14;
	RANK_LEVEL_JOB4 = 15;
	RANK_LEVEL_JOB5 = 16;
	//战力排行
	RANK_FC_TOTAL = 21;
	RANK_FC_JOB1 = 22;
	RANK_FC_JOB2 = 23;
	RANK_FC_JOB3 = 24;
	RANK_FC_JOB4 = 25;
	RANK_FC_JOB5 = 26;
	//装备排行
	RANK_EQUIP_TOTAL = 31;
	RANK_EQUIP_JOB1 = 32;
	RANK_EQUIP_JOB2 = 33;
	RANK_EQUIP_JOB3 = 34;
	RANK_EQUIP_JOB4 = 35;
	RANK_EQUIP_JOB5 = 36;
	//财富排行
	RANK_TREASURE_COIN = 47;
	RANK_TREASURE_GOLD = 48;
	RANK_TREASURE_BIND_GOLD = 49;
	//八卦排行
	RANK_BAGUA_TOTAL = 51;
	RANK_BAGUA_JOB1 = 52;
	RANK_BAGUA_JOB2 = 53;
	RANK_BAGUA_JOB3 = 54;
	RANK_BAGUA_JOB4 = 55;
	RANK_BAGUA_JOB5 = 56;
}

//排行榜玩家信息
message RankPlayerData
{
	required PlayerBaseData baseInfo = 1; //角色基本信息
	required uint32 guildId = 4; //帮会ID
	required string guildName = 5; //帮会名
	required uint32 rankNum = 6; //排名
	required uint32 score = 7; //分数（根据排行榜类型不同，意义不同）
}

//排行榜信息请求
message RankInfoRequest
{
	required uint32 type = 1; //排行榜类型
}

//排行榜信息应答
message RankInfoAnswer
{
	required uint32 result = 1; //结果，0：成功，否则失败
	repeated RankPlayerData infos = 2; //排行榜信息列表
	required uint32 myRank = 3; //我的排名（0表示未上榜）
}







